### PR TITLE
Update local build setup.

### PR DESCRIPTION
**Pull request by dkinzer**
* Allows m1 builds.
* Removes unnecessary build cruft.
* Documents new build configuration options.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,17 +25,22 @@ DEFAULT_RUN_ARGS ?= -e "EXECJS_RUNTIME=Disabled" \
 		-e "TUPRESS_DB_NAME=$(TUPRESS_DB_NAME)" \
 		-e "TUPRESS_DB_PASSWORD=$(TUPRESS_DB_PASSWORD)" \
 		-e "TUPRESS_DB_USER=$(TUPRESS_DB_USER)" \
+		-e "RAILS_LOG_TO_STDOUT=yes" \
 		--rm -it
 
 build:
-	@docker build --build-arg RAILS_MASTER_KEY=$(RAILS_MASTER_KEY) \
+	docker build --build-arg RAILS_MASTER_KEY=$(RAILS_MASTER_KEY) \
```
Comment (sensei100) [inline Makefile:32]: @dkinzer Is the @ symbol no longer needed?

File: .docker/app/Dockerfile [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
     apk add -U --no-cache --virtual build-dependencies \
-      git=2.34.2-r0 \
       build-base=0.5-r2 \
+      git=2.34.2-r0 \
+      libc6-compat \
       libxslt-dev=1.1.35-r0 \
       mariadb-dev=10.6.8-r0 \
       nodejs=16.14.2-r0 \
       yarn=1.22.17-r0 && \
-    gem install bundler:2.3.7  && \
```
Comment (sensei100) [inline .docker/app/Dockerfile:33]: Why don't we need to run bundle install?

Comment (dkinzer) [reply .docker/app/Dockerfile:33]: We still need to run `bundle install` but I did some very thorough debugging of the image and it turns out that even though we are installing bundler:2.3.7 here it doesn't ever get used so this gives a false sense of that is actually happening in the container.  The default bundler that comes with the container is what is actually being used.

**Comment by sensei100** [comment]
@dkinzer I was trying to run this locally, but it has been stuck on installing spring-watcher-listen for about 20 minutes.  Did that happen to you?

**Review by sensei100 — APPROVED** [review]

**Comment by dkinzer** [comment]
Yeah it was slow at that stop but not that slow for me.

**Pull request merged**